### ISSUE TITLE
Cannot set mathlib to ATLAS

### ISSUE THREAD

**Issue by kevininderhees** [kaldi-asr/kaldi#3217]
I'm trying to set mathlib back to ATLAS so that our builds succeed while we explore moving to MKL, but the configure script fails.
```
/opt/kaldi/src# ./configure --shared --mathlib=ATLAS --atlas-root=../tools/ATLAS_headers/
Configuring KALDI to use MKL
Configuring ...
Backing up kaldi.mk to kaldi.mk.bak ...
Checking compiler g++ ...
Checking OpenFst library in /opt/kaldi/tools/openfst ...
Checking cub library in /opt/kaldi/tools/cub ...
Doing OS specific configurations ...
On Linux: Checking for linear algebra header files ...
Configuring MKL library directory: ***configure failed: MKL libraries could not be found. Please use the switch --mkl-libdir  ***
```

I suspect that line 1007 needs to be updated because MKLROOT will always be set from line 813.
```
# If one of these variables is set, we switch the external math library.
is_set $MKLLIBDIR && echo "Configuring KALDI to use MKL" && export MATHLIB="MKL"
is_set $MKLROOT && echo "Configuring KALDI to use MKL"&& export MATHLIB="MKL"
is_set $CLAPACKROOT && echo "Configuring KALDI to use CLAPACK"&& export MATHLIB="CLAPACK"
is_set $OPENBLASROOT && echo "Configuring KALDI to use OPENBLAS"&& export MATHLIB="OPENBLAS"
```

**Comment by danpovey** [comment]
Yeah, @kkm000 is working on a fix, I think there is a PR #3216  already.

**Comment by kkm000** [comment]
@kevininderhees, tangentially, if you have an Intel CPU and a real Linux (not Cygwin, but Windows 10 "Linux subsystem for Windows," sometimes called LSW, probably qualifies), give MKL a go if you can. I've seen comparisons tests with MKL exceeding OpenBLAS speed by the factor of over 3 on a system with AVX512F instructions. Over ATLAS this would  probably translate to 5 if not more. But depends on the  CPU architecture, of course.

If you have no root access, there is a way to pull the debs or rpms and install them to an alternative location, but this is a bit involved, and I do not remember how. But the Internet knows everything.

The fix will be in today.

**Comment by kkm000** [comment]
Another option without root is download a standalone installer here: https://software.intel.com/en-us/mkl/choose-download/linux. Registration required but is not intrusive. I think last time I did that (before they had the package feeds), it asked for the desired installation directory. I went with the default, and it sudo'ed itself, but selecting one you have write access to may do the trick. Then configure with `--mkl-root=` pointing to that directory.
